src/dune_rules/lib.mli

```diff
@@ -131,6 +131,8 @@ module DB : sig
 
   val find_even_when_hidden : t -> Lib_name.t -> lib option Memo.t
 
+  val is_external : t -> Lib_name.t -> bool Memo.t
```

Comment:
Instead of introducing this function. How about just doing a resolve first (with find_even_when_hidden). Once the library is resolved, you can check if it's external by looking at the status.

Reply:
`Status.t` don't give the information if the lib is external or not. And expanding status it's not a good idea, I tried it.
Resolve then a check, could end up with the case when the library is not installed(not found). We want to print the information whether or not the library is installed.

Reply:
The constructors `Installed` and `Installed_private` correspond to external libraries.

Why is expanding the status not a good idea?

> Resolve then a check, could end up with the case when the library is not installed(not found). We want to print the information whether or not the library is installed.

I don't understand. Is there a test case that corresponds to this situation?

Reply:
I said not a good Idea because I tried to have a constructor `External` in the status. It was conflicting whether  the lib is `Found` or `Not_found`.

Yeah I see you suggestion, we doesn't need to expand the status if it's true:
Is it that, if find_even_hidden give `None` the lib is external and if we get a lib, we test whether it's `Installed`?

Reply:
If a lib is not found, we can assume it is an external lib.

Reply:
> Is it that, if find_even_hidden give None the lib is external and if we get a lib, we test whether it's Installed?

Yes, we can do that.

> If a lib is not found, we can assume it is an external lib.

That's the best we can do.

In general, we don't want to modify any existing code in `lib.ml`. It's pretty complicated already and we'd prefer changes that simplify things or at least add things independently. That's why I'm pushing against these changes.

Reply:
I understand, that sounds good for me and it's equivalent to what I was already doing.